network/network.go

```diff
@@ -0,0 +1,65 @@
+package network
+
+import (
+	"golang.org/x/net/context"
+
+	"github.com/dedis/cothority/monitor"
+)
+
+// Conn is the basic interface to represent any communication mean
+// between two host.
+type Conn interface {
+	// Send a message through the connection.
+	// obj should be a POINTER to the actual struct to send, or an interface.
+	// It should not be a Golang type.
+	Send(ctx context.Context, obj Body) error
+	// Receive any message through the connection. It is a blocking call that
+	// returns either when a message arrived or when Close() has been called, or
+	// when a network error occured.
+	Receive(ctx context.Context) (Packet, error)
+	// Close will close the connection. Implementations must take care that
+	// Close() makes Receive() returns with an error, and any subsequent Send()
+	// will return with an error.
+	Close() error
```

Comment:
> What happens when calling Close on an unlistened address or an already closed address?

This is about `Conn` not `Address` so I'm afraid I don't get your question. 
If your question is: `What happens when calling Close on an unlistened address or an already closed Conn?`, the answer is ... undefined ! 
Should we want to put a specific implementation rules that returns an error ? Is that necessary ?
Even though I'm not sure it's necessary, I'd be OK to do that, if you agree.